File: android/src/main/java/com/zh4dev/niimbot_print/Model/PrintLabelModel.java
package com.zh4dev.niimbot_print.Model;

import com.google.gson.Gson;

public class PrintLabelModel {

    String text;
    double fontSize;

    public PrintLabelModel() {
    }

    public PrintLabelModel(String text, double fontSize) {
        this.text = text;
        this.fontSize = fontSize;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public double getFontSize() {
        return fontSize;
    }

    public void setFontSize(double fontSize) {
        this.fontSize = fontSize;
    }

    public String toMap() {
        return new Gson().toJson(this);
    }

    public PrintLabelModel fromMap(String json) {
        return new Gson().fromJson(json, PrintLabelModel.class);
    }
}

File: lib/model/print_label_model.dart
class PrintLabelModel {
  PrintLabelModel({
    String? text,
    num? fontSize,
  }) {
    _text = text;
    _fontSize = fontSize;
  }

  PrintLabelModel.fromJson(dynamic json) {
    _text = json['text'];
    _fontSize = json['fontSize'];
  }

  String? _text;
  num? _fontSize;

  PrintLabelModel copyWith({
    String? text,
    num? fontSize,
  }) =>
      PrintLabelModel(
        text: text ?? _text,
        fontSize: fontSize ?? _fontSize,
      );

  String? get text => _text;

  num? get fontSize => _fontSize;

  Map<String, dynamic> toJson() {
    final map = <String, dynamic>{};
    map['text'] = _text;
    map['fontSize'] = _fontSize;
    return map;
  }
}
